humblegen/src/parser.rs

```diff
@@ -1,21 +1,32 @@
 //! The humble language parser.
 
+mod embeds;
+
 use itertools::Itertools;
 use pest::Parser;
 use pest_derive::Parser;
 
 /// The humble language parser, derived from the PEST grammar file.
 #[derive(Parser)]
 #[grammar = "humble.pest"]
-pub struct HumbleParser;
+struct HumbleParser;
 
 use crate::ast::*;
 
 /// Parse complete spec.
 pub(crate) fn parse(input: &str) -> Result<Spec, pest::error::Error<Rule>> {
-    // TODO: Returns errors proper.
-    let humbled = HumbleParser::parse(Rule::doc, input)?.next().expect("TODO when does this happen?");
-    Ok(Spec(humbled.into_inner().map(parse_spec_item).collect()))
+    let humbled = HumbleParser::parse(Rule::doc, input)?
+        .next()
+        .expect("grammar requires non-empty document");
+
+    let mut ast = Spec(humbled.into_inner().map(parse_spec_item).collect());
+
+    // AST transformations
+    embeds::resolve_embeds(&mut ast);
+
+    // TODO: type checks
```

Comment:
Let's open issues instead of peppering our code with `TODO`s (of which I am the most guilty myself).